design/one-pager-controller-options-overrides.md

```diff
@@ -0,0 +1,42 @@
+# Controller Options Overrides
+
+- Owners:
+  - @max-melentyev
+- Reviewers: Crossplane Maintainers
+- Status: draft
+
+## Background
+
+There are cases when single configuration doesn't work for all controllers of a provider.
+Different Crossplane controllers may have very different load profile and number of managed resources.
+Consider examples in crossplane-contrib/provider-aws:
+
+- Recomciling route53 records is fast, but having many resources of this type may trigger AWS API
```

Comment:
I my head, polling in providers serve two purposes:
* gradually getting a resource from non-existent, to creating and finally ready/synced, and communicating this state and connection parameters to a xr. This is latency sensitive, there's often a human in the loop waiting for this. 
* drift detection and observing and state changes

I think such a change here should take this into consideration. I believe I have read something about it but I'm not on top of all the current activities these days.

Reply:
Purposes are correct. What do you think is missing?

Reply:
One idea is that we add both pollInterval (for resources not ready) and driftInterval (for other resources). But there may be other alternatives.

At least this proposal as it is would allow us to add this interval later without any big change here.

But another question is if you would need per-controller intervals if you have pollInterval and driftInterval as options? Do we need both features, or just one? And if we just have one, which would serve most users/use cases.

Reply:
If it's decided to add a driftInterval, I think it makes sense to have it as a separate option. There can be different scenarios and people may want to response to drift faster than others.